backend/fms_core/services/sample.py

```diff
@@ -602,12 +640,7 @@ def pool_submitted_samples(samples_info,
         errors.append(f"Reception date is not valid.")
 
     if not errors:
-        # Calculate the total volume of the pool
-        pool_volume = sum(sample["volume"] for sample in samples_info)
-        for sample in samples_info:
-            sample_volume = sample["volume"]
-            # Calculate the volume ratio of each sample in the pool
-            sample["volume_ratio"] = decimal_rounded_to_precision(sample_volume / pool_volume, 15)
+        errors.extend(test_and_update_volume_ratio(samples_info))
```

Comment:
Don't use your non service function within a service too. This goes in the importer (or row handler)...